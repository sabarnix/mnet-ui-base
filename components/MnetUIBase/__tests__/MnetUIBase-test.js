"use strict";

var _react = _interopRequireDefault(require("react"));

var _reactTestRenderer = _interopRequireDefault(require("react-test-renderer"));

require("jest-styled-components");

var _react2 = require("@testing-library/react");

var _ = require("..");

var _Heading = require("../../Heading");

var _contexts = require("../../../contexts");

var _mnet = require("../../../themes/mnet");

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { "default": obj }; }

var TestAnnouncer = function TestAnnouncer(_ref) {
  var announce = _ref.announce;

  _react["default"].useEffect(function () {
    return announce('hello', 'assertive');
  });

  return /*#__PURE__*/_react["default"].createElement("div", null, "hi");
};

var customBreakpointsTheme = {
  global: {
    deviceBreakpoints: {
      phone: 'small',
      tablet: 'medium',
      computer: 'large'
    },
    breakpoints: {
      small: {
        value: 600
      },
      medium: {
        value: 800
      },
      large: {
        value: 1000
      }
    }
  }
};

var SSRTester = function SSRTester(_ref2) {
  var ua = _ref2.ua;
  return /*#__PURE__*/_react["default"].createElement(_.MnetUIBase, {
    theme: customBreakpointsTheme,
    userAgent: ua
  }, /*#__PURE__*/_react["default"].createElement(_contexts.ResponsiveContext.Consumer, null, function (size) {
    return /*#__PURE__*/_react["default"].createElement(_Heading.Heading, null, "Received size " + size + " for " + ua);
  }));
};

describe('MnetUIBase', function () {
  afterEach(_react2.cleanup);
  test('basic', function () {
    var component = _reactTestRenderer["default"].create( /*#__PURE__*/_react["default"].createElement(_.MnetUIBase, null));

    expect(component.toJSON()).toMatchSnapshot();
  });
  test('mnet theme', function () {
    var component = _reactTestRenderer["default"].create( /*#__PURE__*/_react["default"].createElement(_.MnetUIBase, {
      theme: _mnet.mnet
    }));

    expect(component.toJSON()).toMatchSnapshot();
  });
  test('themeMode', function () {
    var component = _reactTestRenderer["default"].create( /*#__PURE__*/_react["default"].createElement(_.MnetUIBase, {
      theme: _mnet.mnet,
      themeMode: "dark"
    }));

    expect(component.toJSON()).toMatchSnapshot();
  });
  test('cssVars', function () {
    var component = _reactTestRenderer["default"].create( /*#__PURE__*/_react["default"].createElement(_.MnetUIBase, {
      cssVars: true
    }, "MnetUIBase App"));

    expect(component.toJSON()).toMatchSnapshot();
  });
  test('full', function () {
    var component = _reactTestRenderer["default"].create( /*#__PURE__*/_react["default"].createElement(_.MnetUIBase, {
      full: true
    }, "MnetUIBase App"));

    expect(component.toJSON()).toMatchSnapshot();
  });
  test('background', function () {
    var component = _reactTestRenderer["default"].create( /*#__PURE__*/_react["default"].createElement(_.MnetUIBase, {
      full: true,
      background: "#0000ff"
    }, "MnetUIBase App"));

    expect(component.toJSON()).toMatchSnapshot();
  });
  test('announce', function (done) {
    var _render = (0, _react2.render)( /*#__PURE__*/_react["default"].createElement(_.MnetUIBase, null, /*#__PURE__*/_react["default"].createElement(_contexts.AnnounceContext.Consumer, null, function (announce) {
      return /*#__PURE__*/_react["default"].createElement(TestAnnouncer, {
        announce: announce
      });
    }))),
        container = _render.container;

    expect(container.firstChild).toMatchSnapshot(); // no style, no need for expectPortal

    expect(document.body.querySelector('[aria-live]')).toMatchSnapshot();
    setTimeout(function () {
      // should clear the aria-live container
      expect(document.body.querySelector('[aria-live]')).toMatchSnapshot();
      done();
    }, 600); // wait the aria-live container to clear
  });
  [
  /* eslint-disable max-len */
  'Mozilla/5.0 (iPhone; CPU iPhone OS 9_0_2 like Mac OS X) AppleWebKit/601.1.46 (KHTML, like Gecko) Version/9.0 Mobile/13A452 Safari/601.1 PTST/396', 'Mozilla/5.0 (iPad; CPU OS 11_2_1 like Mac OS X) AppleWebKit/604.4.7 (KHTML, like Gecko) Mobile/15C153 [FBAN/FBIOS;FBAV/156.0.0.41.97;FBBV/89172188;FBDV/iPad5,3;FBMD/iPad;FBSN/iOS;FBSV/11.2.1;FBSS/2;FBCR/;FBID/tablet;FBLC/en_GB;FBOP/5;FBRV/0]', 'Mozilla/5.0 (X11; Linux x86_64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/44.0.2403.157 Safari/537.36'
  /* eslint-enable max-len */
  ].forEach(function (ua) {
    test("ssr rendering " + ua.substring(0, 25), function () {
      var component = _reactTestRenderer["default"].create( /*#__PURE__*/_react["default"].createElement(SSRTester, {
        ua: ua
      }));

      expect(component.toJSON()).toMatchSnapshot();
    });
  });
});