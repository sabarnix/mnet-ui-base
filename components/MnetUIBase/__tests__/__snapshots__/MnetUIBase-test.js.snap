// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MnetUIBase announce 1`] = `
.c0 {
  font-size: 12px;
  line-height: 16px;
  box-sizing: border-box;
  -webkit-text-size-adjust: 100%;
  -ms-text-size-adjust: 100%;
  -moz-osx-font-smoothing: grayscale;
  -webkit-font-smoothing: antialiased;
}

<div
  class="c0"
>
  <div>
    hi
  </div>
</div>
`;

exports[`MnetUIBase announce 2`] = `
<div
  aria-live="assertive"
  style="left: -100%; right: 100%; position: fixed; z-index: -1;"
>
  hello
</div>
`;

exports[`MnetUIBase announce 3`] = `
<div
  aria-live="assertive"
  style="left: -100%; right: 100%; position: fixed; z-index: -1;"
/>
`;

exports[`MnetUIBase background 1`] = `
.c0 {
  font-size: 12px;
  line-height: 16px;
  background-color: #0000ff;
  color: #f8f8f8;
  box-sizing: border-box;
  -webkit-text-size-adjust: 100%;
  -ms-text-size-adjust: 100%;
  -moz-osx-font-smoothing: grayscale;
  -webkit-font-smoothing: antialiased;
  width: 100vw;
  height: 100vh;
  overflow: auto;
}

<div
  className="c0"
>
  MnetUIBase App
</div>
`;

exports[`MnetUIBase basic 1`] = `
.c0 {
  font-size: 12px;
  line-height: 16px;
  box-sizing: border-box;
  -webkit-text-size-adjust: 100%;
  -ms-text-size-adjust: 100%;
  -moz-osx-font-smoothing: grayscale;
  -webkit-font-smoothing: antialiased;
}

<div
  className="c0"
/>
`;

exports[`MnetUIBase cssVars 1`] = `
.c0 {
  font-size: 12px;
  line-height: 16px;
  box-sizing: border-box;
  -webkit-text-size-adjust: 100%;
  -ms-text-size-adjust: 100%;
  -moz-osx-font-smoothing: grayscale;
  -webkit-font-smoothing: antialiased;
  --active: rgba(221,221,221,0.5);
  --active-background: background-contrast;
  --active-text: text-strong;
  --black: #000000;
  --brand: #E15151;
  --focus: #B1C2FE;
  --graph-0: accent-1;
  --graph-1: neutral-1;
  --graph-2: neutral-2;
  --graph-3: neutral-3;
  --graph-4: neutral-4;
  --placeholder: #e0e4ee;
  --selected: dark-2;
  --selected-background: brand;
  --selected-text: text-strong;
  --white: #FFFFFF;
  --accent-1: #38C18B;
  --accent-2: #8F94A6;
  --accent-3: #739FFC;
  --accent-4: #439ADC;
  --dark-1: #313340;
  --dark-2: #4b4d59;
  --dark-3: #65707d;
  --dark-4: #8691a8;
  --dark-5: #999999;
  --dark-6: #999999;
  --light-1: #fbfbfb;
  --light-2: #f6f6f8;
  --light-3: #f5f6f8;
  --light-4: #e1e3ef;
  --light-5: #dfdfdf;
  --light-6: #DADADA;
  --neutral-1: #519bff;
  --neutral-2: #99742E;
  --neutral-3: #00739D;
  --neutral-4: #A2423D;
  --status-critical: #e35e59;
  --status-error: #e35e59;
  --status-warning: #FFAA15;
  --status-ok: #44b88d;
  --status-unknown: #CCCCCC;
  --status-disabled: #CCCCCC;
}

<div
  className="c0"
>
  MnetUIBase App
</div>
`;

exports[`MnetUIBase full 1`] = `
.c0 {
  font-size: 12px;
  line-height: 16px;
  box-sizing: border-box;
  -webkit-text-size-adjust: 100%;
  -ms-text-size-adjust: 100%;
  -moz-osx-font-smoothing: grayscale;
  -webkit-font-smoothing: antialiased;
  width: 100vw;
  height: 100vh;
  overflow: auto;
}

<div
  className="c0"
>
  MnetUIBase App
</div>
`;

exports[`MnetUIBase mnet theme 1`] = `
.c0 {
  font-family: 'Open Sans',Arial,sans-serif;
  font-size: 12px;
  line-height: 16px;
  background-color: #ffffff;
  color: #444444;
  box-sizing: border-box;
  -webkit-text-size-adjust: 100%;
  -ms-text-size-adjust: 100%;
  -moz-osx-font-smoothing: grayscale;
  -webkit-font-smoothing: antialiased;
}

<div
  className="c0"
/>
`;

exports[`MnetUIBase ssr rendering Mozilla/5.0 (X11; Linux x 1`] = `
.c0 {
  font-size: 12px;
  line-height: 16px;
  box-sizing: border-box;
  -webkit-text-size-adjust: 100%;
  -ms-text-size-adjust: 100%;
  -moz-osx-font-smoothing: grayscale;
  -webkit-font-smoothing: antialiased;
}

.c1 {
  font-size: 33.33333333333333px;
  line-height: 37.33333333333333px;
  max-width: 533.3333333333333px;
  font-weight: 600;
}

@media only screen and (max-width:600px) {
  .c1 {
    font-size: 22.666666666666664px;
    line-height: 26.666666666666664px;
    max-width: 362.66666666666663px;
  }
}

<div
  className="c0"
>
  <h1
    className="c1"
  >
    Received size large for Mozilla/5.0 (X11; Linux x86_64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/44.0.2403.157 Safari/537.36
  </h1>
</div>
`;

exports[`MnetUIBase ssr rendering Mozilla/5.0 (iPad; CPU OS 1`] = `
.c0 {
  font-size: 12px;
  line-height: 16px;
  box-sizing: border-box;
  -webkit-text-size-adjust: 100%;
  -ms-text-size-adjust: 100%;
  -moz-osx-font-smoothing: grayscale;
  -webkit-font-smoothing: antialiased;
}

.c1 {
  font-size: 33.33333333333333px;
  line-height: 37.33333333333333px;
  max-width: 533.3333333333333px;
  font-weight: 600;
}

@media only screen and (max-width:600px) {
  .c1 {
    font-size: 22.666666666666664px;
    line-height: 26.666666666666664px;
    max-width: 362.66666666666663px;
  }
}

<div
  className="c0"
>
  <h1
    className="c1"
  >
    Received size medium for Mozilla/5.0 (iPad; CPU OS 11_2_1 like Mac OS X) AppleWebKit/604.4.7 (KHTML, like Gecko) Mobile/15C153 [FBAN/FBIOS;FBAV/156.0.0.41.97;FBBV/89172188;FBDV/iPad5,3;FBMD/iPad;FBSN/iOS;FBSV/11.2.1;FBSS/2;FBCR/;FBID/tablet;FBLC/en_GB;FBOP/5;FBRV/0]
  </h1>
</div>
`;

exports[`MnetUIBase ssr rendering Mozilla/5.0 (iPhone; CPU  1`] = `
.c0 {
  font-size: 12px;
  line-height: 16px;
  box-sizing: border-box;
  -webkit-text-size-adjust: 100%;
  -ms-text-size-adjust: 100%;
  -moz-osx-font-smoothing: grayscale;
  -webkit-font-smoothing: antialiased;
}

.c1 {
  font-size: 33.33333333333333px;
  line-height: 37.33333333333333px;
  max-width: 533.3333333333333px;
  font-weight: 600;
}

@media only screen and (max-width:600px) {
  .c1 {
    font-size: 22.666666666666664px;
    line-height: 26.666666666666664px;
    max-width: 362.66666666666663px;
  }
}

<div
  className="c0"
>
  <h1
    className="c1"
  >
    Received size small for Mozilla/5.0 (iPhone; CPU iPhone OS 9_0_2 like Mac OS X) AppleWebKit/601.1.46 (KHTML, like Gecko) Version/9.0 Mobile/13A452 Safari/601.1 PTST/396
  </h1>
</div>
`;

exports[`MnetUIBase themeMode 1`] = `
.c0 {
  font-family: 'Open Sans',Arial,sans-serif;
  font-size: 12px;
  line-height: 16px;
  background-color: #1f2024;
  color: #f8f8f8;
  box-sizing: border-box;
  -webkit-text-size-adjust: 100%;
  -ms-text-size-adjust: 100%;
  -moz-osx-font-smoothing: grayscale;
  -webkit-font-smoothing: antialiased;
}

<div
  className="c0"
/>
`;
